### PR TITLE
Bring back the `build` directory for temporary builds

### LINKED ISSUE #375: Proposals for fixing the 'dist' issue: publish on a release branch
We could only commit `dist/` on a release branch and publish that instead of master, this is probably the best way to avoid dist diffs showing up in PRs constantly.

For a release we could branch off to say `release/2.2` (or just `release`), when a release is done it could then be tagged (for bower).

### PR DESCRIPTION
**Pull request by parisk**
Fix #375

### CODE REVIEW

**Comment by blink1073** [comment]
Can we also point `main` in `package.json` to this folder, to avoid #356?

**Comment by parisk** [comment]
This is not something we are willing to do at the moment.

If we did so, requiring xterm with `require('xterm');` would not work out of the box and would require building on the library from the consumer. This is not an experience we would like to provide ATM.

If you are using TypeScript you can require each of xterm.js' components relatively from the `src` directory.

**Comment by Tyriar** [comment]
@parisk to clarify, release commits will set `BUILD_DIR`, run the script and check in `dist/`?

**Comment by parisk** [comment]
Yep, exactly. They will set `BUILD_DIR` to `dist` and then do the rest.

**Comment by blink1073** [comment]
I disagree.  In our projects, we have a `src/` folder that has ES6 TypeScript.  We output to a `lib/` directory that is ES5, and our `main` in `package.json` points to the index in that directory.  We ship the `lib/` directory on npm, but keep only `src/` in version control.  We also ship a separate `dist/` directory that could be used for bower.   The `package.json` should point to *node* code, and `bower.json` to the `dist` code.  

cf https://github.com/jupyterlab/services

**Comment by blink1073** [comment]
In this way, consumers require `xterm` in a way that works best for their use case, without needing extra config or importing from `node_modules`.

**Comment by parisk** [comment]
@blink1073 thanks for your comments and proposals.

What you are proposing is change the way we distribute xterm.js. While this is something that can change and can be debated, it is outside the scope of this PR.

The issue that this PR is trying to solve is that when we make temporary builds of xterm.js when we develop xterm.js, this happens in a directory that is being checked into Git by default (`dist`), which is outside the scope of your proposal.

Also, while I understand that your proposal can solve this issue and that it works great for Jupyter, currently I'd like to keep the "importing API" as-is, with all imports/requires etc. requiring files from the `dist` directory **by default**.

**Comment by blink1073** [comment]
Fair enough, I'll move the comment over to #356.

File: bin/build [hunk 1]
```diff
@@ -1,26 +1,35 @@
 #! /usr/bin/env bash
 
+set -e
+
+# $BUILD_DIR should default to "build"
+export BUILD_DIR=${BUILD_DIR:=build}
```
Comment (Tyriar) [inline bin/build]: I don't think you want to export here, that means it will be available outside the scope of the script.

File: bin/build [hunk 2]
```diff
@@ -1,26 +1,35 @@
 #! /usr/bin/env bash
 
+set -e
+
+# $BUILD_DIR should default to "build"
```
Comment (Tyriar) [inline bin/build]: Should we use args instead so we don't need to mess with the env?

**Pull request merged**